xline/src/storage/compact/periodic_compactor.rs

```diff
@@ -109,33 +108,29 @@ impl<C: Compactable> PeriodicCompactor<C> {
             revision, self.period
         );
 
-        let res = self.client.compact(revision).await;
-        if res.is_ok() {
-            info!(
-                "completed auto revision compaction, revision = {}, period = {:?}, took {:?}",
-                revision,
-                self.period,
-                now.elapsed().as_secs()
-            );
-            return target_revision;
-        }
-        if let Err(ClientError::CommandError(ExecuteError::RevisionCompacted(_, compacted_rev))) =
-            res
-        {
-            info!(
-                "required revision {} has been compacted, the current compacted revision is {},  period = {:?}, took {:?}",
-                revision,
-                compacted_rev,
-                self.period,
-                now.elapsed().as_secs()
-            );
-            return Some(compacted_rev);
+        let Some(compactable) = &*self.compactable.read().await else {
+            return None;
+        };
```

Comment:
```rust
let Some(ref compactable) = *self.compactable.read().await else {
    return None;
};
```
try this and remove ` #[allow(clippy::pattern_type_mismatch)]`

Reply:
fixed